Code/GraphMol/AddHs.cpp

```diff
@@ -386,6 +387,13 @@ void addHs(RWMol &mol, bool explicitOnly, bool addCoords,
       mol.addBond(aidx, newIdx, Bond::SINGLE);
       mol.getAtomWithIdx(newIdx)->updatePropertyCache();
       if (addCoords) setHydrogenCoords(&mol, newIdx, aidx);
+      AtomPDBResidueInfo *info = (AtomPDBResidueInfo *)(newAt->getMonomerInfo());
+      if (info && info->getMonomerType() == AtomMonomerInfo::PDBRESIDUE) {
+        AtomPDBResidueInfo *newInfo = new AtomPDBResidueInfo(" H  ", newIdx, "", info->getResidueName(),
```

Comment:
Using the atom index for the serial number might not work in practice as they are not related.  I would find the current maximum serial number and add one to it.

It might be nice is the hydrogens have unique names, especially if one is using/making a refinement dictionary.  In general, this is done by using the number of hydrogen atoms on the residue, i.e. " H1", " H2" again you will need to count the existing hydrogens and names first.

Note that in practice, the hydrogen index "wraps around" due to a mis reading of the initial PDB spec way back in the day.   Hydrogen 123 is written "3H12" for instance.  There was no reason for the H to be in column 1, but someone codified that and now we're stuck :)

The names only have to be unique for the residue, not for the protein.